test/LinkedCellsTest.cpp

```diff
@@ -0,0 +1,219 @@
+
+    auto lc = container::linked_cell<container::index::row_major_index>(
+        {1.0, 1.0, 1.0},
+        1.0,
+        {
+            BoundaryCondition::outflow,
+            BoundaryCondition::outflow,
+            BoundaryCondition::outflow,
+            BoundaryCondition::outflow,
+            BoundaryCondition::outflow,
+            BoundaryCondition::outflow,
+        },
+        1,
+        1.0);
+    lc.insert(Particle({0.5, 0.5, 0.5}, {0, 0, 0}, 0, 0));
+    outflow_pc = container::particle_container(std::move(lc));
+    auto lc2 = container::linked_cell<container::index::row_major_index>(
+        {1.0, 1.0, 1.0},
+        1.0,
+        {
+            BoundaryCondition::reflecting,
+            BoundaryCondition::reflecting,
+            BoundaryCondition::reflecting,
+            BoundaryCondition::reflecting,
+            BoundaryCondition::reflecting,
+            BoundaryCondition::reflecting,
+        },
+        1,
+        0.25);
+    lc2.insert(Particle({0.5, 0.5, 0.5}, {0, 0, 0}, 0, 0));
+    reflecting_pc = container::particle_container(std::move(lc2));
+  }
+};
+
+TEST_F(LinkedCellsTest, test_outflow_left) {
+
+  outflow_pc.linear([](Particle &p) {
+    p.position = {-1.0, 0.5, 0.5};
+  });
+
+  outflow_pc.boundary([](auto p) {});
+  outflow_pc.refresh();
+
+  auto counter = 0;
+  outflow_pc.linear([&counter](auto &p) {
+    counter++;
+  });
+  EXPECT_EQ(counter, 0);
+}
+TEST_F(LinkedCellsTest, test_outflow_rigth) {
+
+  outflow_pc.linear([](Particle &p) {
+    p.position = {2.0, 0.5, 0.5};
+  });
+
+  outflow_pc.boundary([](auto p) {});
+  outflow_pc.refresh();
+
+  auto counter = 0;
+  outflow_pc.linear([&counter](auto &p) {
+    counter++;
+  });
+  EXPECT_EQ(counter, 0);
+}
+TEST_F(LinkedCellsTest, test_outflow_top) {
+
+  outflow_pc.linear([](Particle &p) {
+    p.position = {0.5, 1.5, 0.5};
+  });
+
+  outflow_pc.boundary([](auto p) {});
+  outflow_pc.refresh();
+
+  auto counter = 0;
+  outflow_pc.linear([&counter](auto &p) {
+    counter++;
+  });
+  EXPECT_EQ(counter, 0);
+}
+TEST_F(LinkedCellsTest, test_outflow_bottom) {
+
+  outflow_pc.linear([](Particle &p) {
+    p.position = {0.5, -0.5, 0.5};
+  });
+
+  outflow_pc.boundary([](auto p) {});
+  outflow_pc.refresh();
+
+  auto counter = 0;
+  outflow_pc.linear([&counter](auto &p) {
+    counter++;
+  });
+  EXPECT_EQ(counter, 0);
+}
+
+TEST_F(LinkedCellsTest, test_outflow_front) {
+
+  outflow_pc.linear([](Particle &p) {
+    p.position = {0.5, 0.5, 1.5};
+  });
+
+  outflow_pc.boundary([](auto p) {});
+  outflow_pc.refresh();
+
+  auto counter = 0;
+  outflow_pc.linear([&counter](auto &p) {
+    counter++;
+  });
+  EXPECT_EQ(counter, 0);
+}
+
+TEST_F(LinkedCellsTest, test_outflow_back) {
+
+  outflow_pc.linear([](Particle &p) {
+    p.position = {0.5, 0.5, -1.5};
+  });
+
+  outflow_pc.boundary([](auto p) {});
+  outflow_pc.refresh();
+
+  auto counter = 0;
+  outflow_pc.linear([&counter](auto &p) {
+    counter++;
+  });
+  EXPECT_EQ(counter, 0);
+}
```

Comment:
All of these tests could be combined into one test rather easily, which would avoid duplicating the code so many times.